### PR TITLE
fix(test): fix flaky overfit test

### PR DESCRIPTION
**Pull request by makram93**
Without setting manual seed, in each test runs
- Batch sampler gives different batches
- Model fine-tuned reports different embedding because the weights initialise are always random

This causes the embeddings to be different each time and thus test is flaky

Closes issue #252

### CODE REVIEW

**Review by bwanglzu — APPROVED** [review]
LGTM! even better if we could autouse it in conftest, to make sure every test is running under same seed.

**Review by tadejsv — CHANGES_REQUESTED** [review]
This should be a fixture

**Review by tadejsv — APPROVED** [review]
